### DISCUSSION TITLE
library of  ConsumptionSaving model definitions

### DISCUSSION THREAD

**Discussion by sbenthall** [econ-ark/HARK#1656] [General]
We now have several PRs that address parts of the HARK 1.0 refactoring.
It's gotten to the point where further progress is blocked because we don't have the model information for our current ConsumptionSaving models in a compact Python format. (See e.g. #1296 #1295, as well as #1292, and #1283)

I think the best thing at this point would be to start writing model configuration objects in 'the new style' for the existing ConsumptionSaving models. We could put these in a directory like `HARK/models`. These could then be used for testing generic algorithms.

The hard part is capturing all the nuances that are embedded in the current ConsumptionSaving, and I expect @mnwhite is going to have to ultimately review these.

Naturally, because we don't have functionality that processes these model definitions yet, it will be hard to validate that they are 'correct'; but I think the solution to this chicken-and-egg problem is to irradiate the dinosaur. If we do this, we will learn a lot about what we will need from a future YAML file format.

But it's also something that anybody could work on, and it's even a good entry task for somebody with a structural microeconomics background. (In my opinion, there's no reason why we should add model definitions for other subfields of Economics besides ConsumptionSaving). cc @JohnRGreen

**Comment by mnwhite** [comment]
This is kind of ongoing on several fronts. My expanded simulation framework has model definitions (at least as far as needed for simulation) for a bunch of models, and the rest can be added very easily. Work on the new stage-based representation also continues separately.
